README.md

```diff
@@ -295,16 +392,96 @@ Here is an example:
 (jeison-read 'boolean json-company '(company CEOs 0 founder)) ;; => t
 ```
 
-Unlike arguments to `elt` function, indices can be *negative*. Negative indices have the same semantics as in **Python** language (enumeration from the end):
+Unlike arguments to `elt` function, indices can be *negative*. Negative
+indices have the same semantics as in **Python** language (enumeration from
+the end):
 
 ``` emacs-lisp
 (jeison-read 'string json-company '(company CEOs -1 name last)) ;; => "Johnson"
 (jeison-read 'boolean json-company '(company CEOs -2 cool)) ;; => t
 ```
 
+### Hash tables
+
+There are times when the JSON stream doesn't lend itself to lists or EIEIO
+objects. One such instance is the [CMake][cmake] file API's `reply` attribute:
+
+```json
+	"reply" :
+	{
+		"client-1b4f70c2f678bf78" :
+		{
+			"codemodel-v2" :
+			{
+				"jsonFile" : "codemodel-v2-5b0f688c981e1e250cef.json",
+				"kind" : "codemodel",
+				"version" :
+				{
+					"major" : 2,
+					"minor" : 4
+				}
+			}
+		}
+	}
+```
+
+`"client-1b4f70c2f678bf78"` and `"codemodel-v2"` don't translate cleanly to a
+path or an object -- they function as keys in a hash table.
+
+Here's a partial example of how you use the `jeison-hashtab-of` constraint:
+
+```emacs-lisp
+(jeison-defclass cmake-fapi-reply-client-data ()
+  ((kind :initarg :kind
+         :initform "!!bogus!!"
+         :type string)
+   (version :initarg version
+            :type cmake-fapi-reply-object-version)
+   (jsonFile :initarg jsonFile
+             :initform "invalid-file"
+             :type string)))
+
+(jeison-defclass cmake-fapi-reply ()
+  ((reply :initarg :reply
+          :initform nil
+          :type (or null (jeison-hashtab-of (jeison-hashtab-of cmake-fapi-reply-client-data))))))
+```
+
+The initial value for the `reply` slot is `nil` in this example. To initialize
+the `reply` slot to an empty hash table, you have to add an
+`initialize-instance :after` method, like so:
+
+```emacs-lisp
+
+(jeison-defclass cmake-fapi-reply ()
+  ((reply :initarg :reply
+          :initform nil
+          :type (jeison-hashtab-of (jeison-hashtab-of cmake-fapi-reply-client-data)))))
+
+(cl-defmethod initialize-instance :after ((self cmake-fapi-reply) &rest _args)
+  (setf (oref self reply) (make-hash-table :test 'equal)))
```

Comment:
Why does the user do all that, can we do it automagically if we see the `hash-table` type requirement?

Reply:
Because `:initform` is quotes its arguments.

Reply:
OBE. Example reworked.